test/unit/lib/challenge.test.ts

```diff
@@ -26,3 +27,173 @@ describe('Challenge Generation', (): void => {
     expect(Buffer.byteLength(challenge, 'base64')).toBe(8)
   })
 })
+
+// Each test generates a random key pair
+describe('Signature Verification', (): void => {
+  let challenge: string
+  let signer: crypto.Signer
+
+  beforeEach((): void => {
+    challenge = 'Crypto Auth service Yay!'
+    signer = crypto.createSign('SHA256')
+
+    signer.update(challenge)
+  })
+
+  it('verifies correct signature - EC Key', (): void => {
+    const keyPair = crypto.generateKeyPairSync('ec', {
+      namedCurve: 'secp256k1', // Allowed by FIDO spec
+      publicKeyEncoding: {
+        type: 'spki', // Key infrasructure
+        format: 'pem' // Encoding format
+      },
+      privateKeyEncoding: {
+        type: 'pkcs8',
+        format: 'pem'
+      }
+    })
+
+    const sign = signer.sign(keyPair.privateKey, 'base64')
+
+    expect(verifySign(challenge, sign, keyPair.publicKey)).toEqual(true)
+  })
+
+  // Using another challenge message
+  it('verifies correct signature - EC Key', (): void => {
```

Comment:
there is no need to repeat the test with another challenge, drop it

Reply:
I would, however, consider having some hard coded base-64 characters here that represent a public key, challenge, and signature that you know to be right. Then pump those in and verify that all checks out. The goal here is to avoid a scenario where there is a bug in the challenge generation AND the verification.